csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -402,23 +561,22 @@ public override IArrowArrayStream GetObjects(GetObjectsDepth depth, string? cata
                 {
                     throw new Exception(getTablesResp.Status.ErrorMessage);
                 }
+
+                IReadOnlyDictionary<string, int> columnMap = GetColumnIndexMap(getTablesResp.DirectResults.ResultSetMetadata.Schema.Columns);
                 TRowSet resp = getTablesResp.DirectResults.ResultSet.Results;
 
-                IReadOnlyList<string> catalogList = resp.Columns[0].StringVal.Values;
-                IReadOnlyList<string> schemaList = resp.Columns[1].StringVal.Values;
-                IReadOnlyList<string> tableList = resp.Columns[2].StringVal.Values;
-                IReadOnlyList<string> tableTypeList = resp.Columns[3].StringVal.Values;
+                IReadOnlyList<string> catalogList = resp.Columns[columnMap["TABLE_CAT"]].StringVal.Values;
```

Comment:
Consider creating string constants for columnMap keys.

Reply:
Done